tests/test_optim_optim.py

```diff
@@ -1,47 +1,36 @@
 import unittest
 
 import torch
+from torch.optim.lr_scheduler import StepLR
 import mock
 
 from seq2seq.optim import Optimizer
 
 class TestOptimizer(unittest.TestCase):
 
     def test_init(self):
+        params = [torch.nn.Parameter(torch.randn(2,3,4))]
         try:
-            optimizer = Optimizer(torch.optim.SGD)
+            optimizer = Optimizer(torch.optim.Adam(params))
         except:
             self.fail("__init__ failed.")
 
         self.assertEquals(optimizer.max_grad_norm, 0)
-        self.assertEquals(optimizer.lr_decay, 1)
-        self.assertEquals(optimizer.decay_after_epoch, 0)
-
-    def test_set_parameters(self):
-        learning_rate = 1
-        optim = Optimizer(torch.optim.SGD, lr=learning_rate)
-        params = [torch.nn.Parameter(torch.randn(2,3,4))]
-        optim.set_parameters(params)
-
-        self.assertTrue(type(optim.optimizer) is torch.optim.SGD)
-        self.assertEquals(optim.optimizer.param_groups[0]['lr'], learning_rate)
 
     def test_update(self):
-        optim = Optimizer(torch.optim.SGD,
-                          lr=1,
-                          decay_after_epoch=5,
-                          lr_decay=0.5)
         params = [torch.nn.Parameter(torch.randn(2,3,4))]
-        optim.set_parameters(params)
-        optim.update(0, 10)
-        self.assertEquals(optim.optimizer.param_groups[0]['lr'], 0.5)
+        optimizer = Optimizer(torch.optim.Adam(params, lr=1), max_grad_norm=5)
+        scheduler = StepLR(optimizer.optimizer, 1, gamma=0.1)
+        optimizer.set_scheduler(scheduler)
+        optimizer.update(0, 10)
```

Comment:
Is this supposed to be  `optimizer.update(10, 0)` cause update takes input as `.update(loss, epoch)` ? but I guess it doesn't make a difference to the `lr`